misc/determine-version.py

```diff
@@ -178,15 +178,15 @@ def version_cmp(a, b):
 verbose_print("recent_version = ", recent_version)
 
 
-in_master_branch = False
-git_show_ref =  subprocess.Popen(["git", "show-ref", "master"],
+in_master_branch = True
+git_all_merged_branches = subprocess.Popen(["git", "branch", "-r", "--merged"],
                                  stdout=subprocess.PIPE)
-for line in git_show_ref.stdout:
-    (sha,ref,) = line.split()
-    if (sha == full_rev) and \
-       (ref == "refs/remotes/origin/master" or ref == "refs/remotes/upstream/master"):
-       in_master_branch = True
-       verbose_print("Detected that we are in master branch")
+for line in git_all_merged_branches.stdout:
+    match = re.search('/[0-9]\+\.[0-9]\+\.x', line)
```

Comment:
Looks good, so if it works +1. Just some nitpicks to avoid catching the wrong lines:
* maybe anchor the regex to the end using `$`
* maybe also grep for `upstream` or `origin` to avoid foreign or local branches